client/ayon_hiero/plugins/publish/extract_workfile.py

```diff
@@ -0,0 +1,85 @@
+import os
+import pyblish.api
+
+from ayon_core.pipeline import publish
+
+import hiero
+import tempfile
+
+from qtpy.QtGui import QPixmap
+
+
+class ExtractWorkfile(publish.Extractor):
+    """
+    Extractor export Hiero workfile representation
+    """
+
+    label = "Extract Workfile"
+    order = pyblish.api.ExtractorOrder
+    families = ["workfile"]
+    hosts = ["hiero"]
+
+    def process(self, instance):
+        # create representation data
+        if "representations" not in instance.data:
+            instance.data["representations"] = []
+
+        # asset = instance.context.data["folderPath"]
+        # asset_name = asset.split("/")[-1]
+
+        active_timeline = hiero.ui.activeSequence()
+        # project = active_timeline.project()
+
+        # adding otio timeline to context
+        # otio_timeline = hiero_export.create_otio_timeline()
+        # otio_timeline = instance.data["otioTimeline"]
+
+        # get workfile thumbnail paths
+        tmp_staging = tempfile.mkdtemp(prefix="pyblish_tmp_")
+        thumbnail_name = "workfile_thumbnail.png"
+        thumbnail_path = os.path.join(tmp_staging, thumbnail_name)
+
+        # search for all windows with name of actual sequence
+        _windows = [w for w in hiero.ui.windowManager().windows()
+                    if active_timeline.name() in w.windowTitle()]
+
+        # export window to thumb path
+        QPixmap.grabWidget(_windows[-1]).save(thumbnail_path, 'png')
+
+        # thumbnail
+        thumb_representation = {
+            'files': thumbnail_name,
+            'stagingDir': tmp_staging,
+            'name': "thumbnail",
+            'thumbnail': True,
+            'ext': "png"
+        }
```

Comment:
Not sure if we'd ever run hiero code headless for publishing - but pretty sure this logic would fail in a headless session?

Also, could the timeline ever be active without any window being available to it? If so - then this would also error.

Reply:
I see your point.  but since this code is actually current logic from existing [`precollect_workfile.py`](https://github.com/ynput/ayon-hiero/blob/develop/client/ayon_hiero/plugins/publish/precollect_workfile.py#L43), I've reported into the relevant module.

I'm concerned that changing this logic here might have unexpected/out-of-scope consequences..
Hence I'd rather discuss this as a potential issue in a separate PR, is that fine by you ?

Reply:
Yup - not crucial at all. Just pointing out that it may be problematic in some cases. 👍 Thanks for clarifying.